libnavigator/src/main/java/com/mapbox/navigation/navigator/LocationEx.kt

```diff
@@ -7,18 +7,19 @@ import com.mapbox.navigator.FixLocation
 import java.util.Date
 
 internal fun FixLocation.toLocation(): Location = Location(this.provider).also {
-    it.latitude = this.coordinate.latitude()
-    it.longitude = this.coordinate.longitude()
-    it.time = this.time.time
-    it.speed = this.speed ?: 0f
-    it.bearing = this.bearing ?: 0f
-    it.altitude = this.altitude?.toDouble() ?: 0.0
-    it.accuracy = this.accuracyHorizontal ?: 0f
+    it.latitude = coordinate.latitude()
+    it.longitude = coordinate.longitude()
+    it.time = time.time
+    it.elapsedRealtimeNanos = monotonicTimestampNanoseconds
+    it.speed = speed ?: 0f
+    it.bearing = bearing ?: 0f
+    it.altitude = altitude?.toDouble() ?: 0.0
+    it.accuracy = accuracyHorizontal ?: 0f
```

Comment:
nit: would be better to create a new location object here instead of `?: 0f`

even android location documentation recommends it: https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/location/java/android/location/Location.java#706

Because hasAccuracy will return true, and `0.0f` is a valid accuracy value.

Reply:
``` kotlin
internal fun FixLocation.toLocation(): Location = Location(this.provider).also { location ->
    location.latitude = coordinate.latitude()
    location.longitude = coordinate.longitude()
    location.time = time.time
    location.elapsedRealtimeNanos = monotonicTimestampNanoseconds
    speed?.let { location.speed = it }
    bearing?.let { location.bearing = it }
    altitude?.let { location.altitude = it.toDouble() }
    bearing?.let { location.bearing = it }
    accuracyHorizontal?.let { location.accuracy = it}

    if (isCurrentSdkVersionEqualOrGreaterThan(Build.VERSION_CODES.O)) {
        bearingAccuracy?.let { location.bearingAccuracyDegrees = it }
        speedAccuracy?.let { location.speedAccuracyMetersPerSecond = it }
        verticalAccuracy?.let { location.verticalAccuracyMeters = it }
    }
}
```

Reply:
@kmadsen great catch! 
btw, no need to create a new location, we can just skip value, if it's `null` in `FixLocation` object.
So, in `Location` object `hasValue() == false` and `getValue() == 0.0`. 
Changed  `toLocation()` extension and added `checkLocationWithZeroParams` test.